Cod3rsGrowth.Testes/TestesUnitarios/TesteServicoTenis.cs

```diff
@@ -20,11 +20,25 @@ public TesteServicoTenis()
         }
 
         [Fact]
-        public void Obter_Lista_De_Tenis_Cadastrados()
+        public void Deve_Retornar_Lista_De_Tenis_Cadastrados()
         {
-            var listaDeTenis = _servicoTenis?.ObterTodos();
+            var listaDeTenis = _servicoTenis.ObterTodos();
             Assert.NotNull(listaDeTenis);
             Assert.Equal(4, listaDeTenis.Count);
         }
+
+        [Fact]
+        public void Deve_Retornar_Tenis_Atraves_Do_Id_Informado()
+        {
+            var idTenisProcurado = 0001;
+            idTenisProcurado = _servicoTenis.ObterPorId(Id);
+            Assert.NotNull(idTenisProcurado);
+            Assert.Equal(1, idTenisProcurado);
```

Comment:
Ta, encontrei vários erros aqui.
1 - Você está atribuindo um valor inteiro à variável idTenisProcurado e depois atribuindo o seu objeto que retornou do ObterPorId? Como isso está funcionando?
2- Você está usando essa propriedade id como parâmetro, mas em que momento ele está sendo criado? não vejo em lugar nenhum.
3- nesse Assert.Equal você está verificando se 1 é identico ao idTenisProcurado, que nesse momento deve ser seu objeto que você obteve pelo ObterPorId. Como isso está funcionando também?

Eu posso estar errada, mas esse código parece estar cheio de erros, do tipo que você nem está conseguindo compilar. Se for o caso, você nunca deve pedir PR nessas condições.

Depois de corrigir essas coisas, adiciona mais um assert que comprove para mim que esse objeto que você está buscando é o seu objeto esperado, é de fato o tenis com id 0001.